internal/gatewayapi/securitypolicy.go

```diff
@@ -422,13 +420,21 @@ func (t *Translator) translateSecurityPolicyForRoute(
 			if irListener != nil {
 				for _, r := range irListener.Routes {
 					if strings.HasPrefix(r.Name, prefix) {
-						r.Security = &ir.SecurityFeatures{
-							CORS:          cors,
-							JWT:           jwt,
-							OIDC:          oidc,
-							BasicAuth:     basicAuth,
-							ExtAuth:       extAuth,
-							Authorization: authorization,
+						if errs == nil {
+							r.Security = &ir.SecurityFeatures{
+								CORS:          cors,
+								JWT:           jwt,
+								OIDC:          oidc,
+								BasicAuth:     basicAuth,
+								ExtAuth:       extAuth,
+								Authorization: authorization,
+							}
+						} else {
```

Comment:
should we assign values for all cases ? may help with debugging in logs, it would look like
```
r.Security = ......
if err != nil {
  r.DirectResponse = .....
}
```